synthesizer/benches/utilities/mod.rs

```diff
@@ -0,0 +1,145 @@
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+pub mod benchmark;
+pub use benchmark::*;
+
+pub mod object_store;
+pub use object_store::*;
+
+pub mod workload;
+pub use workload::*;
+
+use console::{
+    account::{PrivateKey, ViewKey},
+    network::Testnet3,
+    prelude::Network,
+    program::{Field, Literal, Plaintext, Record, Value, Zero, U64},
+};
+use snarkvm_synthesizer::{Block, ConsensusStorage, ConsensusStore, Header, Metadata, Transaction, Transition, VM};
+use snarkvm_utilities::TestRng;
+
+use anyhow::Result;
+use indexmap::IndexMap;
+use rand::{CryptoRng, Rng};
+use std::borrow::Borrow;
+
+/// A helper function to initialize a VM with a genesis block.
+pub fn initialize_vm<C: ConsensusStorage<Testnet3>, R: Rng + CryptoRng>(
+    private_key: &PrivateKey<Testnet3>,
+    rng: &mut R,
+) -> (VM<Testnet3, C>, Record<Testnet3, Plaintext<Testnet3>>) {
+    // Initialize the VM.
+    // If the `rocks` feature is enabled, then we are benchmarking using the DB backend.
+    // In this case, we need to create a temporary directory to store the database.
+    // Otherwise, we can just use the in-memory backend.
+    let temp_dir = match cfg!(feature = "rocks") {
+        false => None,
+        true => {
+            let temp_dir = tempfile::tempdir().expect("Failed to open temporary directory").into_path();
+            println!("Using the following temporary directory {:?}", temp_dir);
+            Some(temp_dir)
+        }
+    };
+    let vm = VM::from(ConsensusStore::open_testing(temp_dir).unwrap()).unwrap();
+
+    // Initialize the genesis block.
+    let genesis = Block::genesis(&vm, private_key, rng).unwrap();
+
+    // Fetch the unspent records.
+    let records = genesis.transitions().cloned().flat_map(Transition::into_records).collect::<IndexMap<_, _>>();
+
+    // Select a record to spend.
+    let view_key = ViewKey::try_from(private_key).unwrap();
+    let record = records.values().next().unwrap().decrypt(&view_key).unwrap();
+
+    // Update the VM.
+    vm.add_next_block(&genesis).unwrap();
+
+    (vm, record)
+}
+
+#[allow(unused)]
+/// Construct a new block based on the given transactions.
+pub fn construct_next_block<C: ConsensusStorage<Testnet3>, R: Rng + CryptoRng>(
+    vm: &VM<Testnet3, C>,
+    private_key: &PrivateKey<Testnet3>,
+    transactions: &[Transaction<Testnet3>],
+    rng: &mut R,
+) -> Result<Block<Testnet3>> {
+    // Get the most recent block.
+    let block_hash =
+        vm.block_store().get_block_hash(*vm.block_store().heights().max().unwrap().borrow()).unwrap().unwrap();
+    let previous_block = vm.block_store().get_block(&block_hash).unwrap().unwrap();
+
+    // Construct the new block header.
+    let transactions = vm.speculate(transactions.iter())?;
+    // Construct the metadata associated with the block.
+    let metadata = Metadata::new(
+        Testnet3::ID,
```

Comment:
side note: I wonder why `Metadata` carries `network` at all - it can be derived ad-hoc from its type